packages/core-p2p/lib/monitor.js

```diff
@@ -192,6 +198,23 @@ module.exports = class Monitor {
     return randomPeer
   }
 
+  /**
+   * Get the highest peer.
+   * @return {Peer}
+   */
+  async getHighestPeer () {
```

Comment:
we cannot make guarantee about the height of the peer, because height was updated the last time there was a connection to it.

a better estimation would be height + (currentTime + lastTimeConnection)/blocktime

Reply:
and i would shuffle the filtered list before, also removing the very highest one